### ISSUE TITLE
 THREE.ImageUtils.loadTexture often returns undefined the second time it is run on the same URL

### ISSUE THREAD

**Issue by tremby** [mrdoob/three.js#6779]
The following code *usually* gives me

```
i1: object
i2: undefined
```

```javascript
i1 = THREE.ImageUtils.loadTexture('http://placehold.it/100/100', undefined, function () {
    console.log("i1: " + typeof i1);
    i2 = THREE.ImageUtils.loadTexture('http://placehold.it/100/100', undefined, function () {
        console.log("i2: " + typeof i2);
    });
});
```

I'm testing on Chrome 43 on OS X.

**Comment by dubejf** [comment]
You need to use the callback:

```
var i1, i2;

console.log('step 1');
i1 = THREE.ImageUtils.loadTexture('http://placehold.it/100/100', undefined, function ( tex1 ) {
    console.log("step 3");
    console.log("i1: ", i1, "tex1", tex1);
    i2 = THREE.ImageUtils.loadTexture('http://placehold.it/100/100', undefined, function ( tex2 ) {
        console.log("step 4");
        console.log("i2: ", i2, "tex2", tex2);
    });
    console.log("step 5");
});
console.log('step 2');
```

1. The first call to `loadTexture` returns a texture, stored in `i1`. This texture is empty, `i1.image === 
undefined`.
2. Later, when the image is loaded, the `onLoad` callback is invoked. Here, `tex1 === `i1` and `i1.image !== undefined`.
3. In the second call to `loadTexture`, your image is already loaded. The `onLoad` callback is invoked immediately. `i2` is undefined because `loadTexture` as not returned yet. However, `tex2` is defined.

**Comment by tremby** [comment]
@dubejf: I'm aware that I can use the callback. And I can see that's probably the right way to do it in most cases. That's how I'm working around this.

Maybe I should explain how I was using this. I wasn't nesting another call for loadTexture in the callback as in my example; this was just the shortest illustration of the bug I could think of.

In my code I build a scene, and it may be around on the page for some time and then be destroyed. Later, in the same page load, it may be constructed again. This is done by just running the same code again.

I used this texture in two places (in particular I need its dimensions for some calculations later), so I was storing the result of `loadTexture` in a variable, like the `i1` in the example above. Since I was storing it in a variable anyway, I wasn't using the argument of the callback. For me the callback was just a notification that now the `.image` property is ready. When the callback runs, the variable set by `loadTexture` had been properly populated, and `texture.image` was present and correct and could be used.

But then the scene is destroyed and later it might be made again. If it's made again, the same code runs. Only this time, when `loadTexture` runs it in most cases returns `undefined`. Its callback still runs and so here my code broke, since the variable it was expecting to contain a texture is now undefined. 

Yes, I work around it by setting the variable to null before the `loadTexture` call and then setting that variable to the argument of the callback, so it can be used elsewhere in my code too. That's fine.

This ticket is pointing out that sometimes `loadTexture` returns a perfectly usable object, and other times returns `undefined`. Any way I try to look at it, that's a bug.

**Comment by mrdoob** [comment]
I think this is a dupe of #6572.

**Comment by tremby** [comment]
I'm not convinced after a read through of the other ticket, but possibly.

**Comment by dubejf** [comment]
Fixed by #6798.

**Comment by tremby** [comment]
Thank you!
